packages/react-app/src/components/Network/SwitchNode.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import {
+  Toggler,
+  Modal,
+  ModalDialog,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Typography,
+  Button,
+  Input,
+  Alert,
+} from '@smooth-ui/core-sc'
+
+interface SwitchNodeProps {
+  didUpdateNodeUrl: (nodeUrl: string) => void
+}
+
+interface SwitchNodeState {
+  inputErrorMessage?: string
+}
+
+class SwitchNode extends React.Component<SwitchNodeProps> {
+  state: SwitchNodeState
+
+  constructor(props: SwitchNodeProps) {
+    super(props)
+    this.state = {}
+  }
+
+  public render() {
+    return (
+      <Toggler>
+        {({ toggled, onToggle }) => (
+          <div>
+            <Button variant="info" onClick={() => onToggle(true)}>
+              Switch Node
+            </Button>
+            <Modal opened={toggled} onClose={() => onToggle(false)}>
+              <ModalDialog>
+                <ModalContent>
+                  <ModalHeader>
+                    <Typography variant="h5" m={0}>
+                      Switch node
+                    </Typography>
+                  </ModalHeader>
+                  <ModalBody>
+                    {this.state.inputErrorMessage != undefined ? (
+                      <Alert>{this.state.inputErrorMessage!}</Alert>
+                    ) : (
+                      <div />
+                    )}
+                    <Input
+                      id="inputNode"
+                      size="md"
+                      width="95%"
+                      placeholder="Input node url"
+                    />
+                  </ModalBody>
+                  <ModalFooter>
+                    <Button
+                      variant="success"
+                      onClick={() => this.onSaveNode(onToggle)}
+                    >
+                      Save
+                    </Button>
+                    <Button variant="secondary" onClick={() => onToggle(false)}>
+                      Close
+                    </Button>
+                  </ModalFooter>
+                </ModalContent>
+              </ModalDialog>
+            </Modal>
+          </div>
+        )}
+      </Toggler>
+    )
+  }
+
+  onSaveNode(onToggle: (state?: boolean | undefined) => void) {
+    const inputElement = document.getElementById(
+      'inputNode',
+    ) as HTMLInputElement
+    const inputValue = inputElement.value
+    if (IsURL(inputValue)) {
+      onToggle(false)
+      this.props.didUpdateNodeUrl(inputValue)
+      this.setState({
+        inputErrorMessage: undefined,
+      })
+    } else {
+      this.setState({
+        inputErrorMessage: 'Please enter a valid node address',
+      })
+    }
+  }
+}
+
+function IsURL(str_url: string) {
```

Comment:
A function should be named in `camelCase`.

Reply:
Function param should be in `camelCase` as well.
```suggestion
function ssURL(strUrl: string) {
```